### PR TITLE
Update styling to adapt to new sharing button

### PR DESCRIPTION
**Pull request by esau-morais**
## Description 

What types of changes does your code introduce to links.dev?
> :information_source: _Put a `x` in one or more boxes that might apply to it_

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] Feature (non-breaking change which adds functionality)
- [X] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation (if none of the other choices apply)

### CODE REVIEW

File: user-page/css/base.css [hunk 1]
```diff
@@ -96,13 +105,15 @@
     justify-content: center;
     height: auto;
     position: relative;
-    padding: 16px 20px;
-    width: 100%;
     -webkit-appearance: none;
     -moz-appearance: none;
     appearance: none;
     box-sizing: border-box;
-    vertical-align: middle;
```
Comment (esau-morais) [inline user-page/css/base.css:105]: LSP warned that this property shouldn't be used with `display: block;`

Comment (fatih-yavuz) [reply user-page/css/base.css:105]: If it doesn't have any effect, you can remove it

**Comment by fatih-yavuz** [comment]
Would these changes affect the existing design? If so, how? Can you add a screenshot?

**Comment by esau-morais** [comment]
> Would these changes affect the existing design? If so, how? Can you add a screenshot?

### Changes

This PR applies abstraction to some classes, which affects the following elements:
- [ ] icons
- [ ] buttons

The idea is to divide styling between an icon and button handling with their use cases without implying the `custom.css`

I've applied a default background for the icon button, but it can be forced in the `custom.css` using `!important`

### Screenshots

desktop view

mobile view

**Comment by fatih-yavuz** [comment]
As you mentioned @esau-morais, this is a destructive PR. It forces people to update their custom CSS or their page will look bad. We cannot do it. 
Everything we do must be backward compatible.
Example: 
image

**Comment by esau-morais** [comment]
> As you mentioned @esau-morais, this is a destructive PR. It forces people to update their custom CSS or their page will look bad. We cannot do it. Everything we do must be backward compatible. Example: image

I don't think the user will actually need to update the custom.css once I've only extracted class names in refactor way. How did you reproduce this kind of styling change? Because before I made this PR, I tested with many users' links, including this one

alfreddagenais's links.dev page

**Comment by fatih-yavuz** [comment]
@esau-morais sorry, I tested it without applying the changes for the backend. I will fully test it tomorrow

**Pull request closed without merging**